site/src/pages/RoadmapPage/sidebar/MajorCourseList.tsx

```diff
@@ -37,12 +38,12 @@ interface MajorCourseListProps {
 const MajorCourseList: FC<MajorCourseListProps> = ({ majorWithSpec, onSpecializationChange, selectedSpecId }) => {
   const isDark = useContext(ThemeContext).darkMode;
   const [specsLoading, setSpecsLoading] = useState(false);
-  const [specOptions, setSpecOptions] = useState<{ value: MajorSpecialization; label: string }[]>([]);
   const [resultsLoading, setResultsLoading] = useState(false);
   const [open, setOpen] = useState(true);
 
-  const { major, specialization } = majorWithSpec;
+  const { major, selectedSpec: specialization, specializations } = majorWithSpec;
```

Comment:
Would also suggest renaming the variable here `specialization` to `selectedSpec` since the auto-rename you did in the other file doesn't change the names here.

```suggestion
  const { major, selectedSpec, specializations } = majorWithSpec;
```

Reply:
latest commit should have done that now.  included changes to some instances of specialization to selectedSpec :3